### PR TITLE
Add /heroes/{heroId}/matches route

### PR DESCRIPTION
**Pull request by Luchanso**
It's my first try add odota/ui#1143 /heroes/{heroId}/matches.
Careful, I don't check this code, because I can't configure backend env in docker, maybe later I will check.

Can you help me, with this code?

### CODE REVIEW

File: routes/spec.js [hunk 1]
```diff
@@ -3368,6 +3368,68 @@ Please keep request rate to approximately 3/s.
         },
       },
     },
+    '/heroes/{hero_id}/matches': {
+      post: {
+        summary: 'POST /heroes/{hero_id}/matches',
+        description: 'Most recent games played with this hero',
+        tags: ['request'],
```
Comment (howardchung) [inline routes/spec.js]: tag should be under hero section

File: routes/spec.js [hunk 2]
```diff
@@ -3368,6 +3368,68 @@ Please keep request rate to approximately 3/s.
...
+        responses: {
+          200: {
+            description: 'Success',
+            schema: {
+              type: 'array',
+              items: {
```
Comment (howardchung) [inline routes/spec.js]: I'm creating a matchObject in my teams data PR, we should reuse that

File: routes/spec.js [hunk 3]
```diff
@@ -3368,6 +3368,68 @@ Please keep request rate to approximately 3/s.
...
+                    description: 'duration',
+                    type: 'integer',
+                  },
+                  start_time: {
+                    description: 'start_time',
+                    type: 'integer',
+                  },
+                  win: {
+                    description: 'win',
+                    type: 'string',
```
Comment (howardchung) [inline routes/spec.js]: this is a boolean

File: routes/spec.js [hunk 4]
```diff
@@ -3368,6 +3368,68 @@ Please keep request rate to approximately 3/s.
...
+          const heroId = req.params.hero_id;
+          db.raw(`SELECT
+            matches.match_id,
+            matches.start_time,
+            matches.duration,
+            ((player_matches.player_slot < 128) = matches.radiant_win) win,
+            player_matches.account_id
+            FROM matches
+            JOIN player_matches using(match_id)
+            JOIN heroes on heroes.id = player_matches.hero_id
```
Comment (howardchung) [inline routes/spec.js]: left join just in case the hero doesn't exist/invalid value

**Comment by howardchung** [comment]
I rolled your commit into:
https://github.com/odota/core/pull/1542

**Comment by Luchanso** [comment]
Thank you for review, I understood all

**Pull request closed without merging**